### PR TITLE
Figure out how to implement ObservableObject properly

### PR DESCRIPTION
**Pull request by cobbal**
[ignore-commit-lint]

### CODE REVIEW

File: Core/Core/Store/Store.swift [hunk 1]
```diff
@@ -73,6 +73,19 @@ public class Store<U: UseCase>: NSObject, NSFetchedResultsControllerDelegate {
...
+    @available(iOSApplicationExtension 13.0, *)
+    public var objectWillChange: ObservableObjectPublisher {
+        if let publisher = _objectWillChange as? ObservableObjectPublisher {
+            return publisher
+        }
+        let publisher = ObservableObjectPublisher()
+        _objectWillChange = publisher
+        return publisher
+    }
+
```
Comment (cobbal) [inline Core/Core/Store/Store.swift:88]: Implementing this was the secret to getting ObservableObject working properly

File: Core/Core/Courses/CourseListView.swift [hunk 2]
```diff
@@ -41,33 +39,20 @@ public struct CourseListView: View {
     static var searchBarHeight: CGFloat = UISearchBar().sizeThatFits(.zero).height
 
     public init(allCourses: Store<GetAllCourses>? = nil) {
-        let loadingSubject = CurrentValueSubject<Bool, Never>(true)
-        if let allCourses = allCourses {
-            self.allCourses = allCourses
-            loadingSubject.send(false)
-        } else {
-            self.allCourses = AppEnvironment.shared.subscribe(GetAllCourses()) { store in
```
Comment (cobbal) [inline Core/Core/Courses/CourseListView.swift:49]: Now the `store` parameter in the callback isn't needed anymore, I could go ahead and remove it again

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/gx58bmq) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fgx58bmq)
teacher: [Link](https://rebrand.ly/5bx2k1v) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2F5bx2k1v)
parent: [Link](https://rebrand.ly/j2a5y2q) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fj2a5y2q)

File: Core/Core/Courses/CourseListView.swift (hunk 2 above)
Comment (cobbal) [reply Core/Core/Courses/CourseListView.swift:49]: Went ahead and did it

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please add a reference to a jira ticket. For example: `refs: MBL-10023`
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Please add which apps this change affects. Example: `affects: Teacher, Student` or `affects: none`
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Please add a release note. If no release note is wanted, use `none`. Example: `release note: Fixed a bug that prevented users from enjoying the app.`
  </td>
    </tr>
  </tbody>
</table>



Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 93.31% | 93.31% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/Store/Store.swift (hunk 1 above)
Comment (vanwagonet) [reply Core/Core/Store/Store.swift:88]: nice. Looks like we can just make some things `@Published` when we drop iOS 12, and we won't need this anymore, right?

Comment (cobbal) [reply Core/Core/Store/Store.swift:88]: Yeah, I think so

**Pull request merged**